app/controllers/api/v1/packages_controller.rb

```diff
@@ -225,6 +227,31 @@ def send_stock_item_response
         end
       end
 
+      def add_remove_item
+        render nothing: true, status: 204 and return if params[:quantity].to_i.zero?
+        response = Package::Operations.pack_or_unpack(params, User.current_user.id)
+        if response[:success]
+          render json: { packages_inventories: response[:packages_inventory] }, status: 201
+        else
+          render json: { errors: response[:errors] }, status: 422
+        end
+      end
+
+      def fetch_associated_packages
+        entity = Package.find(params[:id]) if params[:id] # fetch or pallet
```

Comment:
Should we use `find_by(id: params[:id])` ? as on next line we return if `package` is not present.

Reply:
Thanks @swatijadhav , good catch. Yes it should be `find_by` as find will throw exception incase of not found.